lattice/schema_to_json.py

```diff
@@ -415,27 +416,80 @@ def search_for_reference(schema_path : str, subdict : dict) -> bool:
         subdict.update(sub_d)
         subdict.pop('$ref')
         # re-search the substituted dictionary
-        subbed = search_for_reference(schema_path, subdict)
+        subbed = search_for_reference(referenced_schemas, schema_path, subdict)
     else:
         for key in subdict:
             if isinstance(subdict[key], dict):
-                subbed = search_for_reference(schema_path, subdict[key])
+                subbed = search_for_reference(referenced_schemas, schema_path, subdict[key])
             if isinstance(subdict[key], list):
                 for entry in [item for item in subdict[key] if isinstance(item, dict)]:
-                    subbed = search_for_reference(schema_path, entry)
+                    subbed = search_for_reference(referenced_schemas, schema_path, entry)
     return subbed
 
 # -------------------------------------------------------------------------------------------------
-def flatten_json_schema(input_schema, output_dir):
-    '''Generate a flattened schema from a schema with references.'''
-    schema_path = os.path.abspath(os.path.dirname(input_schema))
-    schema = load(input_schema)
-    while search_for_reference(schema_path, schema):
-        pass
-    file_name_root = os.path.splitext(os.path.basename(input_schema))[0]
-    dump(schema, os.path.join(output_dir, file_name_root + '_flat.json'))
+def generate_flat_json_schema(source_schema_input_path, json_schema_output_path):
+    '''Create reference-resolved JSON schema from YAML source schema.'''
+    if os.path.isfile(source_schema_input_path) and '.schema.yaml' in source_schema_input_path:
+        schema_dir = os.path.abspath(os.path.dirname(source_schema_input_path))
+        json_schema_output_dir = os.path.abspath(os.path.dirname(json_schema_output_path))
+        j = JSON_translator()
+        main_schema_instance = j.load_source_schema(source_schema_input_path)
+        schema_ref_map = {'core.schema' : load(os.path.join(json_schema_output_dir, 'core.schema.json'))}
+        for ref_source in os.listdir(schema_dir):
+            schema_ref_map[os.path.splitext(ref_source)[0]] = j.load_source_schema(os.path.join(schema_dir, ref_source))
```

Comment:
Seems cleanest to keep a map local to the pkg function, and populate it in-situ.